plugins/linux-capture/xcomposite-input.cpp

```diff
@@ -0,0 +1,382 @@
+				  a.lex_comparable.end(),
+				  b.lex_comparable.begin(),
+				  b.lex_comparable.end());
+		  });
+
+	for (auto s : window_strings) {
+		obs_property_list_add_string(wins, s.name.c_str(),
+					     s.desc.c_str());
+	}
+
+	obs_properties_add_int(props, "cut_top", obs_module_text("CropTop"), 0,
+			       4096, 1);
+	obs_properties_add_int(props, "cut_left", obs_module_text("CropLeft"),
+			       0, 4096, 1);
+	obs_properties_add_int(props, "cut_right", obs_module_text("CropRight"),
+			       0, 4096, 1);
+	obs_properties_add_int(props, "cut_bot", obs_module_text("CropBottom"),
+			       0, 4096, 1);
+
+	obs_properties_add_bool(props, "swap_redblue",
+				obs_module_text("SwapRedBlue"));
+
+	obs_properties_add_bool(props, "show_cursor",
+				obs_module_text("CaptureCursor"));
+
+	obs_properties_add_bool(props, "include_border",
+				obs_module_text("IncludeXBorder"));
+
+	obs_properties_add_bool(props, "exclude_alpha",
+				obs_module_text("ExcludeAlpha"));
+
+	return props;
+}
+
+void xcompcap_defaults(obs_data_t *settings)
+{
+	obs_data_set_default_string(settings, "capture_window", "");
+	obs_data_set_default_int(settings, "cut_top", 0);
+	obs_data_set_default_int(settings, "cut_left", 0);
+	obs_data_set_default_int(settings, "cut_right", 0);
+	obs_data_set_default_int(settings, "cut_bot", 0);
+	obs_data_set_default_bool(settings, "swap_redblue", false);
+	obs_data_set_default_bool(settings, "show_cursor", true);
+	obs_data_set_default_bool(settings, "include_border", false);
+	obs_data_set_default_bool(settings, "exclude_alpha", false);
+}
+
+void xcompcap_update(void *data, obs_data_t *settings)
+{
+	struct xcompcap *s = (struct xcompcap *)data;
+
+	obs_enter_graphics();
+	pthread_mutex_lock(&s->lock);
+
+	s->crop_top = obs_data_get_int(settings, "cut_top");
+	s->crop_left = obs_data_get_int(settings, "cut_left");
+	s->crop_right = obs_data_get_int(settings, "cut_right");
+	s->crop_bot = obs_data_get_int(settings, "cut_bot");
+	s->swapRedBlue = obs_data_get_bool(settings, "swap_redblue");
+	s->show_cursor = obs_data_get_bool(settings, "show_cursor");
+	s->include_border = obs_data_get_bool(settings, "include_border");
+	s->exclude_alpha = obs_data_get_bool(settings, "exclude_alpha");
+
+	s->windowName = obs_data_get_string(settings, "capture_window");
+	s->win = xcomp_find_window(conn, disp, s->windowName);
+	if (s->win && s->windowName) {
+		blog(LOG_INFO,
+		     "[window-capture: '%s'] update settings:\n"
+		     "\ttitle: %s\n"
+		     "\tclass: %s\n",
+		     obs_source_get_name(s->source),
+		     xcomp_window_name(conn, disp, s->win).c_str(),
+		     xcomp_window_class(conn, s->win).c_str());
+	}
+
+	watcher_register(conn, s);
+	xcomp_cleanup_pixmap(disp, s);
+	xcomp_create_pixmap(conn, disp, s, LOG_ERROR);
+	xcb_xcursor_offset_win(conn, s->cursor, s->win);
+	xcb_xcursor_offset(s->cursor, s->cursor->x_org + s->crop_left,
+			   s->cursor->y_org + s->crop_top);
+
+	obs_leave_graphics();
+	pthread_mutex_unlock(&s->lock);
```

Comment:
I think the preference is to have these in reverse order. This is reflected in the other parts of this PR.

Reply:
Good catch!